src/guides/anatomy-of-an-http-transaction.md

```diff
@@ -0,0 +1,360 @@
+the method and URL, so that appropriate actions can be taken. Node makes this
+relatively painless by putting handy properties onto the `request` object.
+
+```javascript
+var method = request.method;
+var url = request.url;
+```
+> **Note:** The `request` object is an instance of [`IncomingMessage`][].
+
+The `method` here will always be a normal HTTP method/verb. The `url` is the
+full URL without the server, protocol or port, so everything after and including
+the third forward slash in a well-formed URL.
+
+Headers are also not far away. They're in their own object on `request` called
+`headers`.
+
+```javascript
+var headers = request.headers;
+var userAgent = headers['user-agent'];
+```
+
+It's important to note here that all headers are represented in lower-case only,
+regardless of how the client actually sent them. This is helps to simplify the
+task of parsing headers for whatever purpose.
+
+## Request Body
+
+Many HTTP requests, like `POST` or `PUT` for example, might have a body as part
+of the request.
+
+To get at the body data, it's a little different from accessing headers. The
+`request` object that's passed in to a handler implements the
+[`ReadableStream`][] interface. This stream can be listened to or piped elswhere
+just like any other stream. We can grab the data right out of the stream by
+listening to the stream's `'data'` and `'end'` events.
+
+```javascript
+var body = '';
+request.on('data', function(chunk) {
+  body += chunk;
+}).on('end', function() {
+  // at this point, `body` has the entire request body stored in it
+});
+```
+
+> **Note:** This may seem a tad tedious. In many cases, it is. Especially when
+all we want is a nice chunk of data to parse. Luckily, there are modules like
+[`body`][] on [`npm`][] which can help hide away some of this logic. It's
+important to have a good understanding of what's going on before going down that
+road, and that's why you're here!
+
+## A quick thing about Errors
+
+Since the `request` object is a [`ReadableStream`][], it's also an
+[`EventEmitter`][]. As such, it behaves in a slightly different way when an
+error happens.
+
+An error in the `request` stream presents itself by emitting an `'error'` event
+on the stream. **If you don't have a listener for that event, the error will be
+*thrown*, which could crash your Node.js program.** You should therefore add an
+`'error'` listener on your request streams, even if you just log it and
+continue on your way. (Though it's probably best to send some kind of HTTP error
+response. More on that later).
+
+```javascript
+request.on('error', function(err) {
+  // This prints the error message and stack trace to `stderr`.
+  console.error(err.stack);
+});
+```
+
+There are other ways of [handling these errors][] such as using [`domains`][] or
+other abstractions and tools, but always be aware that errors can and do happen,
+and you're going to have to deal with them.
+
+## What we've got so far
```

Comment:
Hmm... the different headers in the doc appear to be using different capitalization approaches. (I like to use http://titlecapitalization.com/ and free my brain from this stuff.)